convert2rhel/subscription.py

```diff
@@ -875,7 +909,7 @@ def needed_subscription_manager_pkgs():
     return to_install_pkgs
 
 
-def _dependencies_to_update(pkg_list):
+def dependencies_to_update(pkg_list):
```

Comment:
```suggestion
```
This function is no longer needed with the direct yum installation. The reason for this function was that the UBI repos had sub-man packages in a version from the latest RHEL minor version and those required dependencies of a version not available in older Alma/Rocky/CentOS minor versions. We now have client tool repos even for older minor versions so this dependency update hack should go away.

It's mentioned in the acceptance criteria of the Jira ticket:

> The use of the new public client-tools repository brings an important improvement - we now have access to sub-man package versions corresponding to specific RHEL minor versions. That was not the case with UBI - there we had access only to the latest version for the particular RHEL major version. With this we shall remove some of the workarounds for errors that stemmed from us installing versions of sub-man intended for a different RHEL minor version:
https://github.com/oamg/convert2rhel/pull/494
https://github.com/oamg/convert2rhel/blob/v1.6.1/convert2rhel/subscription.py#L783 

Also related:
https://github.com/oamg/convert2rhel/pull/905

Reply:
removed